### PR TITLE
fabtests: Fix unexpected msg test to actually test unexpected msgs

### PR DESCRIPTION
**Pull request by shefty**

### CODE REVIEW

File: fabtests/common/shared.c [hunk 1]
```diff
@@ -1656,7 +1656,7 @@ int ft_read_addr_opts(char **node, char **service, struct fi_info *hints,
 {
 	int ret;
 
-	if (opts->dst_addr && (opts->src_addr || !opts->oob_port)){
+	if (opts->dst_addr) {
```
Comment (shefty) [inline fabtests/common/shared.c:1659]: @rwespetal - I believe this change revert is resulting in breaking EFA testing.  I need to understand what EFA is needing, so the correct update can be added.

Comment (rwespetal) [reply fabtests/common/shared.c:1659]: We're using the -E option for out of band setup. Unfortunately in band communication of the addresses doesn't work with EFA + fabtests at the moment.

Comment (shefty) [reply fabtests/common/shared.c:1659]: The -E option makes sense.  The issue I'm seeing is that the original code would call fi_getinfo with FI_SOURCE set, but set node and service to NULL on input.  The test case is calling a fabtest (fi_unexpected_msg) as a client, with only the destination address specified.  That test uses an OOB sync mechanism.

The tests use of an OOB connection should be independent of how the test calls fi_getinfo.  But in the existing code, it is not.

**Comment by rwespetal** [comment]
The NCCL tests are failing to build the psm3 provider, is that build working?

```
09:50:16   CC       prov/psm3/src/libpsm3_la-psmx3_attr.lo
09:50:16 In file included from prov/psm3/src/psmx3.h:75:0,
09:50:16                  from prov/psm3/src/psmx3_am.c:33:
09:50:16 prov/psm3/src/version.h:107:14: error: static declaration of ‘psm2_am_register_handlers_2’ follows non-static declaration
09:50:16  psm2_error_t psm2_am_register_handlers_2(
09:50:16               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
09:50:16 In file included from prov/psm3/src/version.h:39:0,
09:50:16                  from prov/psm3/src/psmx3.h:75,
09:50:16                  from prov/psm3/src/psmx3_am.c:33:
09:50:16 ./prov/psm3/psm3/psm2_am.h:283:14: note: previous declaration of ‘psm2_am_register_handlers_2’ was here
09:50:16  psm2_error_t psm2_am_register_handlers_2(psm2_ep_t ep,
09:50:16               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
09:50:16 Makefile:12233: recipe for target 'prov/psm3/src/libpsm3_la-psmx3_am.lo' failed
09:50:16 make[1]: *** [prov/psm3/src/libpsm3_la-psmx3_am.lo] Error 1
09:50:16 make[1]: *** Waiting for unfinished jobs....
09:50:16 In file included from prov/psm3/src/psmx3.h:75:0,
09:50:16                  from prov/psm3/src/psmx3_atomic.c:33:
09:50:16 prov/psm3/src/version.h:107:14: error: static declaration of ‘psm2_am_register_handlers_2’ follows non-static declaration
09:50:16  psm2_error_t psm2_am_register_handlers_2(
09:50:16               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
09:50:16 In file included from prov/psm3/src/version.h:39:0,
09:50:16                  from prov/psm3/src/psmx3.h:75,
09:50:16                  from prov/psm3/src/psmx3_atomic.c:33:
09:50:16 ./prov/psm3/psm3/psm2_am.h:283:14: note: previous declaration of ‘psm2_am_register_handlers_2’ was here
09:50:16  psm2_error_t psm2_am_register_handlers_2(psm2_ep_t ep,
09:50:16               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
09:50:16 In file included from prov/psm3/src/psmx3.h:75:0,
09:50:16                  from prov/psm3/src/psmx3_attr.c:33:
09:50:16 prov/psm3/src/version.h:107:14: error: static declaration of ‘psm2_am_register_handlers_2’ follows non-static declaration
09:50:16  psm2_error_t psm2_am_register_handlers_2(
09:50:16               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
09:50:16 In file included from prov/psm3/src/version.h:39:0,
09:50:16                  from prov/psm3/src/psmx3.h:75,
09:50:16                  from prov/psm3/src/psmx3_attr.c:33:
09:50:16 ./prov/psm3/psm3/psm2_am.h:283:14: note: previous declaration of ‘psm2_am_register_handlers_2’ was here
09:50:16  psm2_error_t psm2_am_register_handlers_2(psm2_ep_t ep,
09:50:16               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
09:50:16 Makefile:12240: recipe for target 'prov/psm3/src/libpsm3_la-psmx3_atomic.lo' failed
09:50:16 make[1]: *** [prov/psm3/src/libpsm3_la-psmx3_atomic.lo] Error 1
09:50:16 Makefile:12247: recipe for target 'prov/psm3/src/libpsm3_la-psmx3_attr.lo' failed
09:50:16 make[1]: *** [prov/psm3/src/libpsm3_la-psmx3_attr.lo] Error 1
09:50:16 make[1]: Leaving directory '/home/ubuntu/libfabric'
09:50:16 Makefile:4881: recipe for target 'all' failed
```

https://libfabric-ci.aws.a2z.com/job/nccl-multi-node/1663/PROVIDER=efa,label=ubuntu_18.04/console

**Comment by shefty** [comment]
All CI was passing prior to psm3 being merged.  Not sure how the NCCL tests didn't catch this before.

**Comment by shefty** [comment]
If I look at the PR (https://github.com/ofiwg/libfabric/pull/6544), I don't see the AWS CI.  Maybe it didn't run?  Or just doesn't show up after merging?

**Comment by rwespetal** [comment]
I missed adding Adam to the allowed users list for the original PR so it didn't get tested.

**Comment by shefty** [comment]
Ah - because he's new here.  That makes sense.  We were already having a side conversation internally about build issues, so I anticipate a quick fix here.

**Comment by acgoldma** [comment]
I believe this is an error when replacing HAVE_PSM2_... features. In the psm3/configure.m4 they are PSM3 while in code they are PSM2, I will do a quick replace in code, then a large fix to remove the old code no longer needed.

**Comment by rwespetal** [comment]
Sounds good, thanks.

**Comment by acgoldma** [comment]
PSM3 fix is in #6562, it got passed the build of libfabric :)

**Comment by shefty** [comment]
rebased to pick up build fix

**Pull request merged**